### PR TITLE
add libnvme package to wolfi

### PR DESCRIPTION
**Pull request by kranurag7**
packaging this given this is required for https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver incoming package.

### CODE REVIEW

File: libnvme.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    packages:
+      - bash
+      - build-base
+      - gcc
+      - json-c
+      - meson
+      - ninja
+      - openssl
+      - python3
+      - python3-dev
```
Comment (kranurag7) [inline libnvme.yaml:20]: build time dependencies have been take from project readme. (ref: https://github.com/linux-nvme/libnvme?tab=readme-ov-file#prerequisite)

File: libnvme.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  - uses: meson/configure
+
+  - uses: meson/compile
+
+  - uses: meson/install
+
+  - uses: strip
+
+subpackages:
+  - name: libnvme-dev
```
Comment (Dentrax) [inline libnvme.yaml:38]: In Wolfi, it's common practice that most `-dev` packages provides its actual package.

**Comment by kranurag7** [comment]
now all the `: NO` are `: YES` at build time expect one. 

> Run-time dependency json-c found: YES 0.18
> Run-time dependency openssl found: YES 3.4.0
> Header "openssl/opensslv.h" has symbol "LIBRESSL_VERSION_NUMBER" with dependency openssl: NO

Looking at the header `/usr/include/openssl/opensslv.h`, we have `OPENSSL_VERSION_NUMBER` defined but not `LIBRESSL_VERSION_NUMBER` is missing. 

<details><summary> full log details </summary>
<p>

```bash
$ meson setup --wipe .build
The Meson build system
Version: 1.6.0
Source dir: /home/user/libnvme
Build dir: /home/user/libnvme/.build
Build type: native build
Project name: libnvme
Project version: 1.10
C compiler for the host machine: cc (gcc 14.2.0 "cc (Wolfi 14.2.0-r4) 14.2.0")
C linker for the host machine: cc ld.bfd 2.43.1
Host machine cpu family: x86_64
Host machine cpu: x86_64
C++ compiler for the host machine: c++ (gcc 14.2.0 "c++ (Wolfi 14.2.0-r4) 14.2.0")
C++ linker for the host machine: c++ ld.bfd 2.43.1
Found pkg-config: YES (/usr/bin/pkg-config) 2.3.0
Run-time dependency json-c found: YES 0.18
Run-time dependency openssl found: YES 3.4.0
Header "openssl/opensslv.h" has symbol "LIBRESSL_VERSION_NUMBER" with dependency openssl: NO
Has header "openssl/core_names.h" with dependency openssl: YES
Run-time dependency libkeyutils found: YES 1.6.3
Checking if "__builtin_type_compatible_p" compiles: YES
Checking if "typeof" compiles: YES
Checking if "byteswap.h" compiles: YES
Checking if "bswap64" : links: YES
Checking if "statement-expr" compiles: YES
Checking if "isblank" : links: YES
Checking if "linux/mctp.h" compiles: YES
Checking if "netdb" : links: YES
Checking if "ioctl has glibc-style prototype" compiles: YES
Compiler for C supports function attribute fallthrough: YES
Configuring libnvme.spec using configuration
Configuring config.h using configuration
Program python3 found: YES (/usr/bin/python3)
Run-time dependency python found: YES 3.12
Program swig found: YES (/usr/bin/swig)
Has header "Python.h" with dependency python-3.12: YES
Configuring __init__.py using configuration
Configuring nbft-dump-diff.sh using configuration
Configuring gen-nbft-diffs.sh using configuration
Program diff found: YES (/usr/bin/diff)
Program diff found: YES (/usr/bin/diff)
Program tree-diff.sh found: YES (/bin/bash -e /home/user/libnvme/test/sysfs/tree-diff.sh)
Program diff found: YES (/usr/bin/diff)
Program config-diff.sh found: YES (/bin/bash -e /home/user/libnvme/test/config/config-diff.sh)
Configuring conf.py using configuration
Configuring api.rst using configuration
Configuring index.rst using configuration
Configuring quickstart.rst using configuration
Configuring installation.rst using configuration
Configuring mi.rst using configuration
Configuring config-schema.json using configuration
Program /home/user/libnvme/doc/../scripts/kernel-doc found: YES (/home/user/libnvme/doc/../scripts/kernel-doc)
Program /home/user/libnvme/doc/../scripts/kernel-doc-check found: YES (/home/user/libnvme/doc/../scripts/kernel-doc-check)
Build targets in project: 32

libnvme 1.10

  Paths
    prefixdir     : /usr/local
    sysconfdir    : /usr/local/etc
    bindir        : /usr/local/bin
    includedir    : /usr/local/include
    datadir       : /usr/local/share
    mandir        : /usr/local/share/man
    libdir        : /usr/local/lib
    build location: /home/user/libnvme/.build

  Dependencies
    json-c        : true
    OpenSSL       : true
    keyutitls     : true
    libdbus       : false
    Python 3      : true

  Configuration
    git version   : "1.10"

Found ninja-1.9 at /usr/bin/ninja
```

</p>
</details>

File: libnvme.yaml (hunk 2 above)
Comment (kranurag7) [reply libnvme.yaml:40]: thanks for the review, I addressed it via https://github.com/wolfi-dev/os/pull/32228/commits/1186c8ff4bbe2109b07bf4402eddd8983e5b1220

**Review by developer-guy — APPROVED** [review]

**Review by Dentrax — APPROVED** [review]

**Pull request merged**